envoy/network/listener_filter_buffer.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <memory>
+
+#include "envoy/buffer/buffer.h"
+#include "envoy/common/pure.h"
+
+namespace Envoy {
+namespace Network {
+
+/**
+ * Interface for ListenerFilterBuffer
+ */
+class ListenerFilterBuffer {
+public:
+  virtual ~ListenerFilterBuffer() = default;
+
+  /**
+   * Return a single const raw slice to the buffer of the data.
+   * @return a Buffer::ConstRawSlice pointed to raw buffer.
+   */
+  virtual const Buffer::ConstRawSlice rawSlice() const PURE;
+
+  /**
+   * Drain the data from the beginning of the buffer.
+   * @param length the length of data to drain.
+   * @return return True if the drain success, otherwise return False.
+   */
+  virtual bool drain(uint64_t length) PURE;
+
+  /**
+   * Return the length of data in the buffer
+   * @return length The length of data in the buffer.
+   */
+  virtual uint64_t length() const PURE;
```

Comment:
Is this redundant with rawSlice()? Someone can just call rawSlice() and check the length on the slice?

Reply:
OK I don't feel strongly about it, but it seems simple enough to just construct a small struct and then look at the length member which would make this interface method redundant.

Reply:
Got it, I can remove it if that small struct is ok.